### ISSUE TITLE
MLTable date_constant None is not of type 'string'

### ISSUE THREAD

**Issue by martheveldhuis** [Azure/azure-sdk-for-python#30405]
- **azure-ai-ml**: 
- **1.5**: 
- **AzureML compute instance**:
- **python 3.10**: 

**Describe the bug**
When creating an MLTable with a column type datetime, a date constant field gets added with null, which throws an error:

```
Traceback (most recent call last):
  File "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/azure/ai/ml/operations/_data_operations.py", line 299, in create_or_update
    referenced_uris = self._validate(data)
  File "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/azure/ai/ml/_telemetry/activity.py", line 263, in wrapper
    return f(*args, **kwargs)
  File "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/azure/ai/ml/operations/_data_operations.py", line 451, in _validate
    validate_mltable_metadata(
  File "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/azure/ai/ml/_utils/_data_utils.py", line 75, in validate_mltable_metadata
    raise ValidationException(
azure.ai.ml.exceptions.ValidationException: datetime.date_constant: None is not of type 'string'
Invalid MLTable metadata schema:
{
  "type": "string"
}

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  (...) <removed for privacy reasons>
    ml_client.data.create_or_update(dynamic_raw_availability)
  File "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/azure/ai/ml/_telemetry/activity.py", line 263, in wrapper
    return f(*args, **kwargs)
  File "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/azure/ai/ml/operations/_data_operations.py", line 344, in create_or_update
    log_and_raise_error(ex)
  File "/anaconda/envs/azureml_py310_sdkv2/lib/python3.10/site-packages/azure/ai/ml/_exception_helper.py", line 295, in log_and_raise_error
    raise Exception(formatted_error)
Exception: 


1) One or more fields are invalid

Details: 

(x) datetime.date_constant: None is not of type 'string'
Invalid MLTable metadata schema:
{
  "type": "string"
}

Resolutions: 
1) Double-check that all specified parameters are of the correct types and formats prescribed by the Data schema.
If using the CLI, you can also check the full log in debug mode for more details by adding --debug to the end of your command

Additional Resources: The easiest way to author a yaml specification file is using IntelliSense and auto-completion Azure ML VS code extension provides: https://code.visualstudio.com/docs/datascience/azure-machine-learning. To set up VS Code, visit https://docs.microsoft.com/azure/machine-learning/how-to-setup-vs-code
```

Removing this line fixes the issue.

**To Reproduce**
1. Create an MLTable with a datetime column
```
mltable = from_parquet_files(
        paths=paths, include_path_column=False
)
types= {"datetime_call": DataType.to_datetime("%d-%m-%y %H:%M:%S")
}
mltable = mltable.convert_column_types(
      column_types=types
)

mltable.save("...")
```

**Expected behavior**
Either an MLTable like this: 

```
- convert_column_types:
  - column_type:
      datetime:
        date_constant: null
        formats:
        - '%d-%m-%y %H:%M:%S'
    columns: datetime_call
```

Where this doesn't raise any errors.
OR an MLTable like this: 

```
- convert_column_types:
  - column_type:
      datetime:
        formats:
        - '%d-%m-%y %H:%M:%S'
    columns: datetime_call
```
This is my current workaround, to remove the date_constant line which appears to cause the error.

**Comment by kashifkhan** [comment]
Thank you for the feedback @martheveldhuis . We will investigate and get back to you asap. 

cc @azureml-github

**Comment by martheveldhuis** [comment]
Is fixed in version 1.6.0
